### PR TITLE
Add integration test to flag MEF composition breaks

### PR DESCRIPTION
**Pull request by jasonmalinowski**
This will fail until #60494 and #60580 are merged.

### CODE REVIEW

**Review by davidwengier — APPROVED** [review]
Disgusting. I love it.

File: src/VisualStudio/IntegrationTest/New.IntegrationTests/MefCompositionTests.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+                if (line.EndsWith("expected exactly 1 export matching constraints:") &&
+                    StartsWithApplicableSymbol(line))
+                {
+                    var entireError = string.Join(Environment.NewLine, compositionErrorFileLines.Skip(i).TakeWhile(s => !string.IsNullOrWhiteSpace(s)));
+                    compositionErrors.Add(entireError);
+                }
+            }
+
+            Assert.Empty(compositionErrors);
```
Comment (sharwell) [inline src/VisualStudio/IntegrationTest/New.IntegrationTests/MefCompositionTests.cs:44]: The error message produced here is not usable.

```suggestion
            AssertEx.EqualOrDiff("", string.Join(Environment.NewLine, compositionErrors));
```

Comment (jasonmalinowski) [reply src/VisualStudio/IntegrationTest/New.IntegrationTests/MefCompositionTests.cs]: Updated.

**Comment by jasonmalinowski** [comment]
The bugs caught by this have now been merged, so this should hopefully pass now.

**Pull request merged**